src/utilities/strings.ts

```diff
@@ -112,3 +112,80 @@ export const camelCase = (string: string): string => {
       return index === 0 ? match.toLowerCase() : match.toUpperCase()
     })
 }
+
+// Taken from the React escapeTextForBrowser internal utility
+const escapeHtmlRegExp = /["'&<>]/
+
+/**
+ * Escape HTML special characters in the string for output in the browser.
+ *
+ * @param {string} string
+ * @returns {string}
+ */
+export const escapeHTML = (string: string): string => {
+  const match = escapeHtmlRegExp.exec(string)
+
+  if (!match) {
+    return string
+  }
+
+  let escape
+  let html = ''
+  let index
+  let lastIndex = 0
+
+  for (index = match.index; index < string.length; index++) {
+    switch (string.charCodeAt(index)) {
+      case 34: // "
+        escape = '&quot;'
+        break
+      case 38: // &
+        escape = '&amp;'
+        break
+      case 39: // '
+        escape = '&#x27;'
+        break
+      case 60: // <
+        escape = '&lt;'
+        break
+      case 62: // >
+        escape = '&gt;'
+        break
+      default:
+        continue
+    }
+
+    if (lastIndex !== index) {
+      html += string.substring(lastIndex, index)
+    }
+
+    lastIndex = index + 1
+    html += escape
+  }
+
+  return lastIndex !== index ? html + string.substring(lastIndex, index) : html
+}
+
+/**
+ * Autolinks URLs in the given string.
+ *
+ * @param {string} string
+ *
+ * @returns {string}
+ */
+export const autolink = (string: string): string => {
+  const pattern = /(?:(https?:\/\/)|www\d{0,3}\.|www-|[a-z0-9.-]+\.[a-z]{2,4}(?=\/))(?:[^\s()<>]+)*(?:[^\s`!-()\[\]{};:'".,<>?«»“”‘’])/giu
```

Comment:
This matches the `UrlFinder` class in Chat API as much as possible. The only case we don't handle is `()` within the path of a URL. There is some regex used in the PHP implementation that isn't supported in JS.

Reply:
👍 for me! You've added a bunch of test cases. If we run into any strange link parsing issues, we'll look into then and make this util better!